src/handlers/psp.ts

```diff
@@ -349,8 +330,16 @@ const submitTransactions = async (state: State) => {
     []
   );
 
-  // TODO: start new lambdas for each providerSubscriptions array element
-  await processSubscriptions(providerSponsorSubscriptions, baseLogOptions);
+  const providerSponsorPromises = providerSponsorSubscriptions.map(async (providerSponsorSubscription) =>
+    spawn({
+      providerSponsorSubscription,
+      baseLogOptions: baseLogOptions,
+      type: config.nodeSettings.cloudProvider.type,
+    })
+  );
+
+  //TODO: do we want to handle lambda promise rejections separately. e.g. Promise.allSettled().catch()
+  await Promise.all(providerSponsorPromises);
```

Comment:
I was thinking `Promise.allSettled` could be used here instead of `Promise.all` here if we want to wait for some return values or errors?

Reply:
I think that makes sense. @acenolaza @aquarat Any objections?

Reply:
No objections and we could even log errors that might occur for each provider/sponsor promise.

One thing I find weird here is that variable name `providerSponsorSubscription` which includes all subscriptions for a provider/sponsor pair. I feel like `providerSponsorSubscriptions` should be changed to `providerSponsorSubscriptionsList` or `providerSponsorSubscriptionsArray`

Reply:
Agreed @acenolaza  I renamed:
providerSponsorSubscription -> providerSponsorSubscriptions (the type for this was already called ProviderSponsorSubscriptions)
providerSponsorSubscriptions -> providerSponsorSubscriptionsArray